.github/workflows/test.yaml

```diff
@@ -98,5 +98,5 @@ jobs:
       sha: ${{ inputs.sha }}
       package-name: dask_cudf
       # Test against latest dask/distributed/dask-cuda.
-      test-before: "pip install git+https://github.com/dask/dask.git@main git+https://github.com/dask/distributed.git@main git+https://github.com/rapidsai/dask-cuda.git@branch-23.04"
+      test-before: "pip install git+https://github.com/dask/dask.git@main git+https://github.com/dask/distributed.git@main git+https://github.com/rapidsai/dask-cuda.git@branch-23.06"
```

Comment:
Is this automated correctly? It looks like line 104 of `update-version.sh` should've fixed this. Here's raft's code, for comparison:

https://github.com/rapidsai/raft/blob/22ebc72e8c0c2738a30a0bbf9b0efc52736eddd6/ci/release/update-version.sh#L64-L67

Reply:
It is automated correctly in cudf:

https://github.com/rapidsai/cudf/blob/branch-23.04/ci/release/update-version.sh#L101

I verified running it locally and the script should've upgraded this automatically but looks like somehow it wasn't.

Reply:
Looks like this is a BSD sed vs GNU sed issue. When I run this script during burndown, I use my Mac which uses BSD sed and it misses this change. When I updated to `gsed` in the `update-version.sh` script it worked as expected.

Let me think about how to handle this in the future.

Reply:
I assume (though cannot confirm since I don't use a Mac) that the issue with that expression lies within this string: `branch-[^\"\s]\+`

Therefore we can probably just change this:
```sh
for FILE in .github/workflows/*.yaml; do
  sed_runner "s/dask-cuda.git@branch-[^\"\s]\+/dask-cuda.git@branch-${NEXT_SHORT_TAG}/g" ${FILE};
done
```

to this:

```sh
for FILE in .github/workflows/*.yaml; do
  sed_runner "s/dask-cuda.git@branch-[0-9][0-9].[0-9][0-9]/dask-cuda.git@branch-${NEXT_SHORT_TAG}/g" ${FILE};
done
```